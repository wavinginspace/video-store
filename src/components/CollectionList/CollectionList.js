import React, { Component } from 'react';
import CollectionLink from '../CollectionLink/CollectionLink';
import './CollectionList.scss';

export class CollectionList extends Component {
  static defaultProps = {
    collections: []
  };

  render() {
    const { collections } = this.props;
    return (
      <>
        <h2 className="collections-header">Your Collections</h2>
        <section className="CollectionList">
          <ul className="collection-list box" aria-live="polite">
            {collections.map(collection => (
              <CollectionLink key={collection.id} collection={collection} />
            ))}
          </ul>
        </section>
      </>
    );
  }
}

export default CollectionList;
