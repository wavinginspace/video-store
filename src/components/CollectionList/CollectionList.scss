@import '../../styles/_breakpoints.scss';
@import '../../styles/_variables.scss';

.CollectionList {
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
}

.collection-list {
  margin: 0;
  padding: 0;
}

.collections-header {
  margin-bottom: 1rem;
}

p, li, h2, a {
  font-family: 'Roboto Mono';
}

