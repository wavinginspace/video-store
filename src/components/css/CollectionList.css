.CollectionList {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
      -ms-flex-pack: center;
          justify-content: center;
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
}

.collection-list {
  margin: 0;
  padding: 0;
}

.collections-header {
  margin-bottom: 1rem;
}
/*# sourceMappingURL=CollectionList.css.map */