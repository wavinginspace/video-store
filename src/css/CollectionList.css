:root {
  --main: var(--main);
  --nav: #499acc;
  --headergrad: linear-gradient(0deg, #ff7676, #fffd8f);
  --light: #fffd8f;
  --headerog: #ff7676;
}

.CollectionList {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
      -ms-flex-pack: center;
          justify-content: center;
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
}

.collection-list {
  margin: 0;
  padding: 0;
}

.collections-header {
  margin-bottom: 1rem;
}

p, li, h2, a {
  font-family: 'Roboto Mono';
}
/*# sourceMappingURL=CollectionList.css.map */