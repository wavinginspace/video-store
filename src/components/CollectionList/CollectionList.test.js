import React from 'react';
import ReactDOM from 'react-dom';
import { MemoryRouter as Router } from 'react-router-dom';
import CollectionList from './CollectionList';
import renderer from 'react-test-renderer';

it('renders without crashing', () => {
  // first create a DOM element to render the component into
  const div = document.createElement('div');

  // render the component, this is the actual test, if something is wrong it will fail here
  ReactDOM.render(<CollectionList collections={[]} />, div);

  // clean up code
  ReactDOM.unmountComponentAtNode(div);
});

it('renders the UI as expected', () => {
  const tree = renderer.create(<CollectionList collections={[]} />).toJSON();
  expect(tree).toMatchSnapshot();
});
