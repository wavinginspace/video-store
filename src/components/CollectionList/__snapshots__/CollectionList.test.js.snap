// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
Array [
  <h2
    className="collections-header"
  >
    Your Collections
  </h2>,
  <section
    className="CollectionList"
  >
    <ul
      aria-live="polite"
      className="collection-list box"
    />
  </section>,
]
`;
